### PR TITLE
TpBot: Add core module

### PR DESCRIPTION
**Pull request by Mecharyry**
#### Problem
We started adding a lot of code into the `mobile` module when it is not directly tied to `android`

#### Solution
Create a `core` module and move non-android specific classes. Additionally create the `Services` and `Views` for the `Human` and `Bot` parts of the app. These will be implemented in the next PR as part of the presenter layer.

### CODE REVIEW

File: TelepresenceBot/core/src/main/java/com/novoda/tpbot/support/Observable.java [hunk 1]
```diff
@@ -15,7 +13,7 @@ protected Observable() {
 
     public synchronized Observable<T> attach(Observer<T> observer) {
         if (observer == null) {
-            throw new DeveloperError("You cannot attach a null observer");
+            throw new NullPointerException("You cannot attach a null observer");
```
Comment (Mecharyry) [inline TelepresenceBot/core/src/main/java/com/novoda/tpbot/support/Observable.java]: Didn't have access to utils here and didn't want to add it as it isn't really necessary.

File: TelepresenceBot/core/src/main/java/com/novoda/tpbot/bot/BotView.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package com.novoda.tpbot.bot;
+
+import com.novoda.tpbot.Direction;
+
+interface BotView {
```
Comment (Mecharyry) [inline TelepresenceBot/core/src/main/java/com/novoda/tpbot/bot/BotView.java]: The plan is to use this in the follow-up presenters PR for the bot part of the app.

File: TelepresenceBot/core/src/main/java/com/novoda/tpbot/bot/BotTpService.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.novoda.tpbot.bot;
+
+import com.novoda.tpbot.Direction;
+import com.novoda.tpbot.Result;
+import com.novoda.tpbot.support.Observable;
+
+interface BotTpService {
```
Comment (Mecharyry) [inline TelepresenceBot/core/src/main/java/com/novoda/tpbot/bot/BotTpService.java]: The plan is to use this in the follow-up presenters PR for the bot part of the app.

Comment (danybony) [inline TelepresenceBot/core/src/main/java/com/novoda/tpbot/bot/BotTpService.java]: It's not very clear how you will use the listen part, but ok for now, let's see in the next PRs

File: TelepresenceBot/core/src/main/java/com/novoda/tpbot/bot/BotView.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+    void onConnect(String message);
+
+    void onError(String message);
+
+    void moveIn(Direction direction);
```
Comment (ataulm) [inline TelepresenceBot/core/src/main/java/com/novoda/tpbot/bot/BotView.java]: I foresee many accidents 😂

File: TelepresenceBot/core/src/main/java/com/novoda/tpbot/human/HumanTpService.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
+package com.novoda.tpbot.human;
+
+import com.novoda.tpbot.Direction;
+import com.novoda.tpbot.Result;
+import com.novoda.tpbot.support.Observable;
+
+interface HumanTpService {
```
Comment (ataulm) [inline TelepresenceBot/core/src/main/java/com/novoda/tpbot/human/HumanTpService.java]: I see now why `BotTpService`

File: TelepresenceBot/mobile/build.gradle [hunk 6]
```diff
@@ -32,7 +32,5 @@ dependencies {
     compile 'com.novoda:notils-java:3.0.2'
```
Comment (ataulm) [inline TelepresenceBot/mobile/build.gradle]: move notils-java to the core project?

File: TelepresenceBot/core/build.gradle [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+
+dependencies {
+    compile fileTree(dir: 'libs', include: ['*.jar'])
+    testCompile 'junit:junit:4.12'
+    testCompile 'org.mockito:mockito-core:1.10.19'
+    testCompile 'org.mockito:mockito-all:1.10.19'
+}
+
+sourceCompatibility = "1.7"
+targetCompatibility = "1.7"
```
Comment (Mecharyry) [inline TelepresenceBot/core/build.gradle]: Now you have seen it with speech marks 👅 this file was automatically generated like this. As I am such a nice guy I have changed it to appease your OCD 👅 	c654bdd

File: TelepresenceBot/core/src/main/java/com/novoda/tpbot/support/Observable.java (hunk 1 above)
Comment (ataulm) [inline TelepresenceBot/core/src/main/java/com/novoda/tpbot/support/Observable.java]: I retracted my 👍  - it's already added in this project (so no difference in APK) just in the wrong module. Moving it to core would mean both modules can use it so "didn't have access" wouldn't be valid.

**Pull request merged**